Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -151,12 +151,39 @@ class UpstreamIntegrationTests: XCTestCase {
         wait(for: [edgeRequestContentExpectation, networkResponseExpectation], timeout: asyncTimeout)
     }
     
-    func test_testFailureExample() {
-        let experienceEvent = ExperienceEvent(xdm: ["xdmtest": "data"],
-                                              data: ["data": ["test": "data"]])
+    func testJSONComparisonSystem() {
+        let multilineValidation = #"""
+          {
+            "integerCompare": 456,
+            "decimalCompare": 123.123,
+            "stringCompare": "abc",
+            "boolCompare": true,
+            "nullCompare": null,
+            "arrayCompare": [1,2,3],
+            "dictionaryCompare": {
+              "nested1": "value1"
```

Comment:
we should also test the recursivity by having multi-level nested dictionaries and arrays with JSON objects

Reply:
Now this one looks like a good candidate to be moved to [AnyCodableTests.swift](https://github.com/adobe/aepsdk-core-ios/blob/main/AEPServices/Tests/utility/AnyCodableTests.swift)

Reply:
I was referring to create a separate test class AnyCodableUtilsTests that test the new util helpers you have added assertEqual(..) and since these functions do the asserts internally you will need to expect a failure for invalid JSONs.